base/timegm.cpp

```diff
@@ -0,0 +1,57 @@
+#include "timegm.hpp"
+
+// There are issues with this implementation due to absence
+// of time_t fromat specification. There are no guarantees
+// of its internal structure so we cannot rely on + or -.
+
+// It could be possible to substitute time_t with boost::ptime
+// in contexts where we need some arithmetical and logical operations
+// on time and we don't have to use precisely time_t.
+
+namespace
+{
+// Number of days elapsed since Jan 01 up to each month
+// (except for February in leap years).
+static int const monoff[] = {
+  0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334
+};
+
+bool IsLeapYear(int year)
+{
+  return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+}
+
+int LeapDaysCount(int y1, int y2)
+{
+  --y1;
+  --y2;
+  return (y2/4 - y1/4) - (y2/100 - y1/100) + (y2/400 - y1/400);
+}
+} // namespace
+
+namespace base
+{
+// Inspired by python's calendar.py
+time_t TimeGM(std::tm *tm)
```

Comment:
clang-format please. BTW, if you'll pass tm by reference there will not be questions like "how this function behave when null is passed as an argument."